src/Microsoft.Health.Fhir.Shared.Api/Controllers/DataConvertController.cs

```diff
@@ -0,0 +1,156 @@
+    [ValidateModelState]
+    public class DataConvertController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly ILogger _logger;
+        private readonly DataConvertConfiguration _config;
+        private static Dictionary<string, HashSet<string>> _supportedParams = InitSupportedParams();
+
+        public DataConvertController(
+            IMediator mediator,
+            IOptions<OperationsConfiguration> operationsConfig,
+            ILogger<DataConvertController> logger)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(operationsConfig?.Value?.DataConvert, nameof(operationsConfig));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _mediator = mediator;
+            _config = operationsConfig.Value.DataConvert;
+            _logger = logger;
+        }
+
+        [HttpPost]
+        [Route(KnownRoutes.DataConvert)]
+        [AuditEventType(AuditEventSubType.DataConvert)]
+        public async Task<IActionResult> DataConvert([FromBody] Parameters inputParams)
+        {
+            CheckIfDataConvertIsEnabled();
+
+            ValidateParams(inputParams);
+
+            string inputData = ReadStringParameter(inputParams, JobRecordProperties.InputData);
+            string templateSetReference = ReadStringParameter(inputParams, JobRecordProperties.TemplateSetReference);
+            string entryPointTemplate = ReadStringParameter(inputParams, JobRecordProperties.EntryPointTemplate);
+            DataConvertInputDataType inputDataType = ReadEnumParameter<DataConvertInputDataType>(inputParams, JobRecordProperties.InputDataType);
+
+            var dataConvertRequest = new DataConvertRequest(inputData, inputDataType, templateSetReference, entryPointTemplate);
+            DataConvertResponse response = await _mediator.Send(dataConvertRequest, cancellationToken: default);
+
+            return new ContentResult
+            {
+                Content = response.Resource,
+                ContentType = "application/json",
+            };
+        }
+
+        private void ValidateParams(Parameters inputParams)
+        {
+            if (inputParams == null)
+            {
+                _logger.LogInformation("Failed to deserialize data convert request body as Parameters resource.");
+                throw new RequestNotValidException(Resources.DataConvertParametersNotValid);
+            }
+
+            var supportedParams = _supportedParams[Request.Method];
+
+            foreach (var param in inputParams.Parameter)
+            {
+                var paramName = param.Name;
+                if (!supportedParams.Contains(paramName))
+                {
+                    throw new RequestNotValidException(string.Format(Resources.DataConvertParameterNotValid, paramName));
+                }
+            }
+        }
+
+        private static string ReadStringParameter(Parameters parameters, string paramName)
```

Comment:
These are two useful methods. We don't do much with Parameters currently, but I could see us needing these in more places in the future. For now I think they are fine in this file, but we might want to move them to a Parameter parser file at some point in the future.

Reply:
Agree.